test/nodes/test_shaper.py

```diff
@@ -0,0 +1,563 @@
+from haystack import Pipeline, Document
+
+
+def test_basic_function_invocation(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      func: expand
+                      output: questions
+                      params:
+                        expand_target: query
+                        size:
+                          func: len
+                          params:
+                            - documents
+                    documents:
+                      func: concat_docs
+                      output: documents
+                      params:
+                        docs: documents
+                        delimiter: " "
+              type: Shaper
+            pipelines:
+              - name: query
+                nodes:
+                  - name: shaper
+                    inputs:
+                      - Query
```

Comment:
I believe we have to use `- Query` here as input due to how we set up our `Pipeline.load_from_yaml` function, but it is a little confusing because it looks like we are missing the `Documents` input which we defined for the `shaper` above. I'm not sure if we can do anything about this, but I thought I would mention it.

Reply:
Lots of things are confusing with the current pipeline design. Refactored design is coming soon. Will keep you in the loop @sjrl